bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/BrokenLinksReport.java

```diff
@@ -0,0 +1,211 @@
+     *     <li>An html fragment containing links , e.g,
+     *     <pre>
+     *       &lt;p&gt;
+     *         &lt;a href="/content/site/page.html"&gt;hello&lt;/a&gt;
+     *         &lt;img src="/content/dam/image1.png"&gt;hello&lt;/a&gt;
+     *       &lt;/p&gt;
+     *     </pre>
+     *     </li>
+     * </ol>
+     *
+     * @param property an entry from a ValueMap
+     * @param htmlFields  lst of properties containing html
+     * @return stream containing extracted references
+     */
+    static Stream<String> collectPaths(Map.Entry<String, Object> property, Set<String> htmlFields) {
+        Object p = property.getValue();
+
+        Stream<String> stream;
+        if (p.getClass() == String[].class) {
+            stream = Arrays.stream((String[]) p);
+        } else if (p.getClass() == String.class){
+            stream = Stream.of((String) p);
+        } else {
+            stream = Stream.empty();
+        }
+        if (htmlFields.contains(property.getKey())) {
+            stream = stream.flatMap(val -> {
+                try {
+                    // parse html and extract links via underlying tagsoup library
+                    LinkContentHandler linkHandler = new LinkContentHandler();
+                    HtmlParser parser = new HtmlParser();
+                    parser.parse(new ByteArrayInputStream(val.getBytes("utf-8")), linkHandler, new Metadata(), new ParseContext());
+                    return linkHandler.getLinks().stream().map(Link::getUri);
+                } catch (Exception e) {
+                    return Stream.empty();
+                }
+            });
+        }
+        return stream;
+    }
+
+    /**
+     * Collect broken references from properties of the given resource
+     *
+     * @param resource      the resource to check
+     * @param regex         regex to to detect properties containing references. Set from @FormField
+     * @param skipList      properties to ignore. Set from @FormField
+     * @param htmlFields    field containing html .
+     * @return broken references keyed by property. The value is a List because a property can contain multiple links,
+     * e.g. if it is multivalued or it is html containing multiple links.
+     */
+    static Map<String, List<String>> collectBrokenReferences(Resource resource, Pattern regex, Set<String> skipList, Set<String> htmlFields) {
+
+        return resource.getValueMap().entrySet().stream()
+                .filter(entry -> !skipList.contains(entry.getKey()))
+                .collect(Collectors.toMap(
+                        entry -> resource.getPath() + "/" + entry.getKey(),
+                        entry -> {
+                            List<String> brokenPaths =  collectPaths(entry, htmlFields)
+                                    .filter(href -> regex.matcher(href).matches())
+                                    .filter(path -> ResourceUtil.isNonExistingResource(resource.getResourceResolver().resolve(path)))

```

Comment:
Is there actually a need to use `ResourceResolver.resolve()` here vs. `ResourceResolver.getResource()`? I would expect that links in content are to the actual resource path, not the mapped path.

Reply:
I wrote this tool to detect potential 404s which can come from mapped paths as well. For completeness sake I would like to keep ResourceResolver.resolve()